kamon-stackdriver/src/main/scala/kamon/stackdriver/StackdriverSpanReporter.scala

```diff
@@ -65,66 +59,42 @@ class StackdriverSpanReporter extends SpanReporter {
       logger.error("Failed to close TraceServiceClient", error)
     }
 
-  private def writeTraces(traces: Traces): Unit = {
-    val request = PatchTracesRequest
-      .newBuilder()
-      .setProjectId(projectId)
-      .setTraces(traces)
-      .build()
+  private def convertSpan(span: FinishedSpan): Span = span match {
+    case FinishedSpan(context, operationName, start, end, tags, _) =>
+      val traceId = context.traceID.string
+      val spanId  = context.spanID.string
+      val name    = SpanName.of(projectId, traceId, spanId).toString
+      Span
+        .newBuilder()
+        .setName(name)
+        .setDisplayName(TruncatableString.newBuilder().setValue(operationName).build())
+        .setStartTime(instantToTimestamp(start))
+        .setEndTime(instantToTimestamp(end))
+        .setAttributes(Attributes.newBuilder().putAllAttributeMap(tagsToLabels(tags).asJava))
+        .setSpanId(spanId)
+        .setParentSpanId(context.parentID.string)
+        .build()
+  }
 
-    client.patchTracesCallable.futureCall(request).onComplete {
-      case Success(_) => //ok
+  private def writeSpans(spans: Seq[Span]): Unit = {
+    val projectName = ProjectName.of(projectId).toString
```

Comment:
I know that builder doesn't accept ProjectName type, but this looks strange:-/ Maybe it would be better to create helper which will build that identifier directly as string without creating instance of `ProjectName`, also probably this could be created once in `configure` method.